tools/db_stress.cc

```diff
@@ -792,17 +792,32 @@ class SharedState {
 
     printf("Choosing random keys with no overwrite\n");
     Random rnd(seed_);
-    size_t num_no_overwrite_keys = (max_key_ * FLAGS_nooverwritepercent) / 100;
+    // Start with the identity permutation. Subsequent iterations of
+    // for loop below will start with perm of previous for loop
+    int64_t *permutation = new int64_t[max_key_];
+    for (int64_t i = 0; i < max_key_; i++) {
+      permutation[i] = i;
+    }
+
     for (auto& cf_ids : no_overwrite_ids_) {
+      // Now do the Knuth shuffle
+      for (int64_t i = max_key_ - 1; i > 0; i--) {
```

Comment:
you could iterate in the reverse order and only up until `max_key_ * FLAGS_nooverwritepercent / 100` to save a bit more time

Reply:
Good call, this saves you around 5-7 secs per iteration.

Also, found a bug. Turns out you can't do the clever XOR-ing to swap, because if you land on an index that doesn't move (i == rand_index), then it points to the same spot in memory, and then the XOR always results in 0.